File: src/main/java/io/codeka/gaia/registries/gitlab/GitLabRawContent.java
package io.codeka.gaia.registries.gitlab;

import io.codeka.gaia.registries.RegistryRawContent;

import java.util.regex.Pattern;

public class GitLabRawContent extends RegistryRawContent {
    @Override
    protected final Pattern getPattern() {
        return Pattern.compile("^(http[s]?://[www.]?gitlab.*).git$");
    }

    @Override
    protected String getTemplateRawUrl() {
        return "{0}/raw/{1}";
    }
}

File: src/main/java/io/codeka/gaia/registries/gitlab/GitLabRawContent.kt
package io.codeka.gaia.registries.gitlab

import io.codeka.gaia.registries.RegistryRawContent
import io.codeka.gaia.registries.RegistryType
import org.springframework.stereotype.Component
import org.springframework.web.client.RestTemplate

@Component
class GitLabRawContent(restTemplate: RestTemplate) : RegistryRawContent(RegistryType.GITLAB, restTemplate)